src/javascript-file.ts

```diff
@@ -0,0 +1,288 @@
+    return new JavascriptDataStructure(body);
+  }
+  constructor(private readonly body: unknown) {
+    super("JSDataToken");
+  }
+  stringify(level = 0, idt = "  ") {
+    return javascriptStringify(this.body, level, idt) ?? "undefined";
+  }
+}
+
+interface JavascriptDependenciesOptions {
+  cjs?: boolean;
+}
+export class JavascriptDependencies extends CodeResolvableBase {
+  imports: Map<string, Array<string>> = new Map();
+  defaultImports: Map<string, string> = new Map();
+  froms = new Set<string>();
+  cjs: boolean;
+  constructor(props: JavascriptDependenciesOptions = {}) {
+    super("JSDependenciesToken");
+
+    this.cjs = props.cjs ?? false;
+  }
+  addImport(
+    imports: Array<string> | string,
+    from: string
+  ): Array<JavascriptRaw> {
+    if (typeof imports === "string") {
+      if (this.defaultImports.has(from)) {
+        if (this.defaultImports.get(from) === imports) {
+          return [JavascriptRaw.value(imports)];
+        }
+        throw new Error(
+          `Default import already exists for ${from}: ${this.defaultImports.get(
+            from
+          )} !== ${imports}`
+        );
+      }
+      this.defaultImports.set(from, imports);
+      this.froms.add(from);
+      return [JavascriptRaw.value(imports)];
+    }
+    if (this.imports.has(from)) {
+      this.imports.set(from, this.imports.get(from)!.concat(imports));
+    } else {
+      this.imports.set(from, imports);
+    }
+    this.froms.add(from);
+    return imports.map((i) => JavascriptRaw.value(i));
+  }
+  stringify(level: number, _idt: string) {
+    if (level !== 0) {
+      throw new Error("JavascriptDependencies cannot be nested");
+    }
+    const out: Array<string> = [];
+    for (const from of this.froms) {
+      const imports = this.imports.get(from) ?? [];
+      const defaultImport = this.defaultImports.get(from);
+      const value = imports.map((i) => i.toString()).join(", ");
+      if (defaultImport && imports.length > 0) {
+        if (this.cjs) {
+          out.push(`const ${defaultImport} = require('${from}');`);
+          out.push(`const { ${value} } = ${defaultImport};`);
+        } else {
+          out.push(`import ${defaultImport}, { ${value} } from '${from}';`);
+        }
+      } else if (defaultImport) {
+        if (this.cjs) {
+          out.push(`const ${defaultImport} = require('${from}');`);
+        } else {
+          out.push(`import ${defaultImport} from '${from}';`);
+        }
+      } else {
+        if (this.cjs) {
+          out.push(`const { ${value} } = require('${from}');`);
+        } else {
+          out.push(`import { ${value} } from '${from}';`);
+        }
+      }
+    }
+    return out.join("\n");
+  }
+}
+
+/**
+ * Options for the JsConfigFile class.
+ */
+export interface JavascriptFileOptions
+  extends JavascriptDependenciesOptions,
+    JsonFileOptions {}
+
+/**
+ * Represents a JS configuratin file (e.g. .eslintrc.js).
+ * Suppor
+ */
+export class JavascriptFile extends JsonFile {
```

Comment:
Can we get rid of the dependency on `JsonFile`. It doesn't make much sense semantically. What features we actually gaining from it?

```suggestion
export class JavascriptFile extends FileBase {
```